app/models/manageiq/providers/redhat/infra_manager/event_parser.rb

```diff
@@ -206,4 +209,27 @@ def self.parse_new_resource_pool(cluster)
 
     rp_hash
   end
+
+  def self.hash_to_ostruct(hash)
+    OpenStruct.new(
+      hash.each_with_object({}) do |(key, val), memo|
+        memo[key] = val.kind_of?(Hash) ? hash_to_ostruct(val) : val
+      end
+    )
+  end
+
+  def self.ovirtobj_to_hash(obj)
+    obj.instance_variables.each_with_object({}) do |k, h|
+      val = obj.instance_variable_get(k)
+
+      h[k.to_s[1..-1]] = case val
+                         when OvirtSDK4::Identified
```

Comment:
I assume this is the base class for anything else?

Reply:
Yes according to the docs is the base for every common OvirtSDK4 obj (I assume is the base for all that we can find in events)
https://www.rubydoc.info/gems/ovirt-engine-sdk/4.3.0/OvirtSDK4/Identified

Reply:
:+1: